### PR TITLE
Eliminate extraneous bool in DatePicker haptic feedback

### PR DESCRIPTION
**Pull request by cbracken**

### CODE REVIEW

**Comment by abarth** [comment]
LGTM

**Comment by abarth** [comment]
Thanks!

File: packages/flutter/lib/src/material/date_picker.dart [hunk 1]
```diff
@@ -603,26 +601,34 @@ class _DatePickerDialogState extends State<_DatePickerDialog> {
...
+    switch (Theme.of(context).platform) {
+      case TargetPlatform.android:
+      case TargetPlatform.fuchsia:
+        HapticFeedback.vibrate();
+        break;
+      case TargetPlatform.iOS:
+        break;
+    }
+  }
+
```
Comment (Hixie) [inline packages/flutter/lib/src/material/date_picker.dart]: If we find any other places where we put this switch, we should probably consider factoring it out into the theme or some such.

Comment (cbracken) [inline packages/flutter/lib/src/material/date_picker.dart]: Good point. We've got something similar-ish in `TimePicker` now. I'd imagine that further to #6574, there'll be others. It's also possible that more granularity to the HapticFeedback API could solve this, with certain types of feedback triggering only on Android, and perhaps a generic `vibrate()`.

Android defines a [set of constants](https://developer.android.com/reference/android/view/HapticFeedbackConstants.html) for the various forms of haptic feedback supported on that platform. iOS is [a bit more generic](https://developer.apple.com/reference/uikit/uifeedbackgenerator?language=objc) (`Notification`, `Impact`, `Selection`).

**Pull request merged**